core/src/main/java/com/adobe/aem/commons/assetshare/components/details/impl/TagsImpl.java

```diff
@@ -88,10 +91,21 @@
         final String[] tagIds = asset.getProperties().get(tagPropertyName, new String[]{});
 
         for (final String tagId : tagIds) {
-            final Tag tag = tagManager.resolve(tagId);
-
-            if (tag != null) {
-                overrideTagTitles.add(tag.getTitle(locale));
+        	final Tag tag = tagManager.resolve(tagId);
+
+        	if (tag != null) {
+        		overrideTagTitles.add(tag.getTitle(locale));
+        	}
+        	// Added the logic for showing the smart tags as part of the tags component. 
+        	// Also removed some unused imports from the class.
+        	else if (StringUtils.isNotBlank(tagId) && showFreeTextProperty) {
+            	List<String> listOfComputedPropValues = new ArrayList<>();
```

Comment:
Simplify this else condition. Should just be:
overrideTagTitles.add(tagId);

Reply:
In else we do not have overrideTagTitles.add(tagId);